test/Controllers/ResourcesTest.php

```diff
@@ -0,0 +1,230 @@
+     */
+    public function testGetResourceById() {
+        $createResourceBody = $this->resourceData('create');
+
+        // Create GonebusyLib\Models\EntitiesResourceResponse:
+        $responseResource = $this->resources->createResource(Configuration::$authorization, $createResourceBody);
+
+        $response = $this->resources->getResourceById(
+            Configuration::$authorization,
+            $responseResource->resource->id
+        );
+
+        // Was it created?
+        $this->assertInstanceOf('GonebusyLib\Models\GetResourceByIdResponse', $response);
+
+         // Does it have all the original data we sent?
+        $responseBody = $this->dataFromResponse($response, 'create');
+        $this->assertEquals($responseBody, $createResourceBody);
+
+        // Delete test resource:
+        $delResponse = $this->resources->deleteResourceById(Configuration::$authorization, $responseResource->resource->id);
+    }
+
+    /**
+     * Test PUT /resources/{id}}
+     * GonebusyLib\Controllers\ResourceController::updateResourceById()
+     */
+    public function testUpdateResourceById() {
+        $createResourceBody = $this->resourceData('create');
+        $createResponse = $this->resources->createResource(Configuration::$authorization, $createResourceBody);
+
+        $anotherResourceBody = $this->resourceData('update');
+
+        // Update the same user:
+        $response = $this->resources->updateResourceById(
+            Configuration::$authorization,
+            $createResponse->resource->id,
+            $anotherResourceBody);
+
+        // Was it fetched?
+        $this->assertInstanceOf('GonebusyLib\Models\UpdateResourceByIdResponse', $response);
+
+        // Does it have all the new data we sent?
+        $responseBody = $this->dataFromResponse($response, 'update');
+        $this->assertEquals($responseBody, $anotherResourceBody);
+    }
+
+    /**
+     * Test GET /resources
+     * Assumes there's at least 3 Resources in the API's database.
+     * GonebusyLib\Controllers\ResourcesController::getResources()
+     */
+    public function testGetResources() {
+        $perPage = 3;
+        $response = $this->resources->getResources(
+            Configuration::$authorization,
+            $page = 1,
+            $perPage);
+
+        // Was it fetched?
+        $this->assertInstanceOf('GonebusyLib\Models\GetResourcesResponse', $response);
+
+        // Did it return an array of 3 resources?
+        $this->assertCount($perPage, $response->resources); // Slightly hardcoded to assume $resources exists in $response.
+        foreach($response->resources as $resource) {
+            $this->assertInstanceOf('GonebusyLib\Models\GetResourcesResponse', $response);
+        }
+    }
+
+    /**
+     * Test GET /resources/things
+     * XXX Assumes there's 0 Things in the API's database.
+     * Test GonebusyLib\Controllers\ResourcesController::getResourceThings()
```

Comment:
The only Q related to this PR @alexagranov, is that there's no things coming from the API so this test isn't gonna do much for now, in fact it's hardcoded to expect to receive an empty array.